ci/test/utils/fetch-merge-target.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+#
+# This requires BUILDKITE_REPO_REF to be set.
```

Comment:
To make it work locally could default to origin/main?

Reply:
✅ Refactored with https://github.com/MaterializeInc/materialize/pull/19157/commits/a71a82c48e9ed21e3b67ab36f121731017d857a8.